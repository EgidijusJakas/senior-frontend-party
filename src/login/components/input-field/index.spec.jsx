import React from 'react';
import { shallow } from 'enzyme';
import { InputField } from '.';

const defaultProps = {
    onChange: jest.fn(),
    placeholder: 'Placeholder',
    value: ''
};

describe.only('Input field', () => {
    it('should render input field', () => {
        const renderedComponent = shallow(<InputField {...defaultProps} />);
        expect(renderedComponent).toMatchSnapshot();
    });
    it('should render input field with error', () => {
        const renderedComponent = shallow(<InputField {...defaultProps} errorMessage="Value cannot be empty" />);
        expect(renderedComponent).toMatchSnapshot();
    });
});
