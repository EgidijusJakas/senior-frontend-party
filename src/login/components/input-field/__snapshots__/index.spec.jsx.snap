// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input field should render input field 1`] = `
<styled.div>
  <styled.input
    className="form-control"
    onChange={[MockFunction]}
    placeholder="Placeholder"
    type="text"
    value=""
  />
</styled.div>
`;

exports[`Input field should render input field with error 1`] = `
<styled.div>
  <styled.input
    className="form-control"
    errorMessage="Value cannot be empty"
    onChange={[MockFunction]}
    placeholder="Placeholder"
    type="text"
    value=""
  />
  <ErrorMessage
    message="Value cannot be empty"
  />
</styled.div>
`;
